tests/Nervos/SignerTests.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright © 2017-2021 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "HexCoding.h"
+#include "Nervos/Address.h"
+#include "Nervos/Signer.h"
+#include "PrivateKey.h"
+#include "PublicKey.h"
+
+#include <gtest/gtest.h>
+
+using namespace TW;
+using namespace TW::Nervos;
+
+TEST(NervosSigner, Sign) {
+    auto signingInput = SigningInput();
+    signingInput.toAddress = "ckb1qzda0cr08m85hc8jlnfp3zer7xulejywt49kt2rr0vthywaa50xwsqfq0njqh7wm5"
+                             "69txsf85n5du4hvqwsqmdsl7zz3s";
+    signingInput.changeAddress = "ckb1qzda0cr08m85hc8jlnfp3zer7xulejywt49kt2rr0vthywaa50xwsqds6ed78"
+                                 "yze6eyfyvd537z66ur22c9mmrgz82ama";
+    signingInput.amount = 10000000000;
+    signingInput.byteFee = 1;
+    auto cell1 = Cell();
+    cell1.outPoint.txHash =
+        parse_hex("b8a5e232c78cd25c1af9b0c00f93867ac7b2a18312c1b0a0e8311a5dcca250e4");
+    cell1.outPoint.index = 0;
+    cell1.capacity = 20000000000;
+    cell1.lock = Script(Address("ckb1qzda0cr08m85hc8jlnfp3zer7xulejywt49kt2rr0vthywaa50xwsqwyk5x9er"
+                                "g8furras980hksatlslfaktks7epf25"));
+    signingInput.cells.push_back(cell1);
+    signingInput.privateKeys.push_back(
+        PrivateKey(parse_hex("8a2a726c44e46d1efaa0f9c2a8efed932f0e96d6050b914fde762ee285e61feb")));
+    auto tx = Transaction();
+    ASSERT_EQ(tx.plan(signingInput), Common::Proto::OK);
+    ASSERT_EQ(tx.sign(signingInput.privateKeys), Common::Proto::OK);
+    ASSERT_EQ(hex(tx.hash()), "d575198d229071f32774b8d718c2f40a68db837919973d88bfa89f441c180173");
```

Comment:
Have you tried sending real world tx signed by wallet core to network?

Reply:
Yes, we had tested our code with real transactions. After doing so many modifications, we did a fresh set of transactions and also updated test cases in accordance with the transactions that we did recenty.